docs/how-to/hip_runtime_api/asynchronous.rst

```diff
@@ -0,0 +1,487 @@
+executed asynchronously. Multiple streams may point to the same device and
+those streams may be fed from multiple concurrent host-side threads. Execution
+on multiple streams may be concurrent but isn't required to be.
+
+Managing streams
+-------------------------------------------------------------------------------
+
+Streams enable the overlap of computation and data transfer, ensuring
+continuous GPU activity. 
+
+To create a stream, the following functions are used, each returning a handle
+to the newly created stream:
+
+- :cpp:func:`hipStreamCreate`: Creates a stream with default settings.
+- :cpp:func:`hipStreamCreateWithFlags`: Allows creating a stream, with specific
+  flags, listed below, enabling more control over stream behavior:
+
+  - ``hipStreamDefault``: creates a default stream suitable for most
+    operations. It ensures that the stream is not non-blocking.
+  - ``hipStreamNonBlocking``: creates a non-blocking stream, allowing
+    concurrent execution of operations. It ensures that tasks can run
+    simultaneously without waiting for each other to complete, thus improving
+    overall performance.
+
+- :cpp:func:`hipStreamCreateWithPriority``: Allows creating a stream with a
+  specified priority, enabling prioritization of certain tasks.
+
+The :cpp:func:`hipStreamSynchronize` function is used to block the calling host
+thread until all previously submitted tasks in a specified HIP stream have
+completed. It ensures that all operations in the given stream, such as kernel
+executions or memory transfers, are finished before the host thread proceeds.
+
+.. note::
+
+  If the :cpp:func:`hipStreamSynchronize` function input stream is 0 (or the
+  default stream), it waits for all operations in the default stream to
+  complete.
+
+Concurrent execution between host and device
+-------------------------------------------------------------------------------
+
+Concurrent execution between the host (CPU) and device (GPU) allows the CPU to
+perform other tasks while the GPU is executing kernels. Kernels can be launched
+asynchronously using ``hipLaunchKernelDefault`` with a stream, enabling the CPU
+to continue executing other code while the GPU processes the kernel. Similarly,
+memory operations like :cpp:func:`hipMemcpyAsync` can be performed
+asynchronously, allowing data transfers between the host and device without
+blocking the CPU.
+
+Concurrent kernel execution
+-------------------------------------------------------------------------------
+
+Concurrent execution of multiple kernels on the GPU allows different kernels to
+run simultaneously to maximize GPU resource usage. Managing dependencies
+between kernels is crucial for ensuring correct execution order. This can be
+achieved using :cpp:func:`hipStreamWaitEvent`, which allows a kernel to wait
+for a specific event before starting execution.
+
+Independent kernels can only run concurrently, if there are enough registers
```

Comment:
```suggestion
Independent kernels can only run concurrently if there are enough registers
```
This isn't clear to me. Does it mean that independent kernels can only run concurrently? Or they can only run concurrently if there are enough resources?

Reply:
They can only run concurrently if there are enough resources.
(Independent kernel can run sequentially too)

Comma is removed